stdparams.go

```diff
@@ -0,0 +1,38 @@
+// +build go1.7
+
+package httprouter
+
+import (
+	"context"
+	"net/http"
+)
+
+type paramsKey struct{}
```

Comment:
@Thomasdezeeuw I am aware of the benefits of defining a `paramsKey` type in the package. I am just arguing that there is no actual benefit of making this a `struct{}` instead of an `int`. You are just saving the size of an int times the number of instances of the type (here always exactly 1) which in this case is so small that it feels just a bit like cargo cult (no offense). I agree with @teepark that this is only a minor complaint but I found it is worth pointing out in a review and getting the opinion about this from other people.

The only downside I see with the `struct{}` approach is that this is arguably more confusing to beginners and it diverges from the canonical example and practice in other libraries.